lib/session.go

```diff
@@ -14,6 +15,8 @@ import (
 	"golang.org/x/crypto/ssh/agent"
 )
 
+var validRoleArn = regexp.MustCompile("arn:aws:iam::\\d+:role/\\w+")
```

Comment:
This doesn't allow for partitions other than "aws". We might want to just use `ParseARN()` instead and validate the requisite components (`iam` and `role/` specifically).

Additionally, role names and paths are allowed to have more than just `\w+`, so this regex would fail for some roles.